### PR TITLE
contrib/gajim: new package (1.9.1)

### PR DESCRIPTION
**Pull request by ttyyls**

### CODE REVIEW

File: contrib/gajim/template.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    "python-emoji",
+    "python-gobject",
+    "python-keyring",
+    "python-nbxmpp",
+    "python-omemo-dr",
+    "python-packaging",
+    "python-pillow",
+    "python-qrcode",
+    "python-sqlalchemy",
+    "python-typing_extensions",
```
Comment (nekopsykose) [inline contrib/gajim/template.py:29]: patch this one out of depends

File: contrib/python-emoji/template.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+pkgver = "2.12.1"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools",
+    "python-wheel",
+]
+checkdepends = ["python-pytest", "python-typing_extensions"]
```
Comment (nekopsykose) [inline contrib/python-emoji/template.py:11]: typing_extensions is an actual dependency and not checkdepends (that you should patch out)

File: contrib/python-jaraco.context/template.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+pkgver = "5.3.0"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools_scm",
+    "python-wheel",
+]
+depends = ["python-tempora"]
```
Comment (nekopsykose) [inline contrib/python-jaraco.context/template.py:11]: this dependency is not imported anywhere here

File: contrib/python-jaraco.functools/template.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+pkgrel = 0
+build_style = "python_pep517"
+make_check_args = ["--deselect=test_functools.py"]  # unpackaged deps
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools_scm",
+    "python-wheel",
+]
+depends = ["python-more-itertools", "python-jaraco.classes"]
```
Comment (nekopsykose) [inline contrib/python-jaraco.functools/template.py:12]: this depends only on more-itertools

File: contrib/python-keyring/template.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools_scm",
+    "python-wheel",
+]
+checkdepends = ["python-pytest"]
+depends = [
+    # XXX: probably should replaced with a global keyring provider
+    "gnome-keyring",
```
Comment (nekopsykose) [inline contrib/python-keyring/template.py:14]: as a stopgap, replace it with nothing

File: contrib/python-keyring/template.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    "python-jaraco.functools",
+]
+pkgdesc = "Store and access your passwords safely"
+maintainer = "ttyyls <contact@behri.org>"
+license = "MIT"
+url = "https://github.com/jaraco/keyring"
+source = f"$(PYPI_SITE)/k/keyring/keyring-{pkgver}.tar.gz"
+sha256 = "daaffd42dbda25ddafb1ad5fec4024e5bbcfe424597ca1ca452b299861e49f1b"
+# unpackaged dependencies
+options = ["!check"]
```
Comment (nekopsykose) [inline contrib/python-keyring/template.py:25]: seems easy to skip the bad ones

File: contrib/python-more-itertools/template.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+pkgname = "python-more-itertools"
+pkgver = "10.3.0"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python-build",
+    "python-flit_core",
+    "python-installer",
+    "python-setuptools",
```
Comment (nekopsykose) [inline contrib/python-more-itertools/template.py:9]: setuptools/wheel don't have to be here if flit_core is used

File: contrib/python-portend/template.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+pkgver = "3.2.0"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools_scm",
+    "python-wheel",
+]
+checkdepends = ["python-pytest", "python-tempora", "python-jaraco.functools"]
```
Comment (nekopsykose) [inline contrib/python-portend/template.py:11]: tempora is a direct dependency and functools is not imported by anything here

File: contrib/python-qrcode/template.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools",
+    "python-wheel",
+]
+depends = ["python-pypng"]
+checkdepends = depends + [
+    "python-pytest",
+    "python-typing_extensions",
```
Comment (nekopsykose) [inline contrib/python-qrcode/template.py:14]: this is a direct dep

File: contrib/python-sqlalchemy/template.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+hostmakedepends = [
+    "python-build",
+    "python-cython",
+    "python-installer",
+    "python-setuptools",
+    "python-wheel",
+]
+checkdepends = [
+    "python-pytest",
+    "python-typing_extensions",
```
Comment (nekopsykose) [inline contrib/python-sqlalchemy/template.py:14]: this is a direct dep

File: contrib/python-tempora/template.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+build_style = "python_pep517"
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools_scm",
+    "python-wheel",
+]
+checkdepends = [
+    "python-pytest",
+    "python-jaraco.functools",
```
Comment (nekopsykose) [inline contrib/python-tempora/template.py:13]: functools and pytz are direct deps

**Comment by nekopsykose** [comment]
could you check the dependency lists? you get this wrong for basically every python thing every time, and i'm not sure how that's possible unless you add random stuff into checkdepends until it passes tests

**Comment by ttyyls** [comment]
> could you check the dependency lists? you get this wrong for basically every python thing every time, and i'm not sure how that's possible unless you add random stuff into checkdepends until it passes tests

true, I'll make sure to read through the imports later and reduce them

File: contrib/python-portend/template.py (hunk 8 above)
Comment (ttyyls) [reply contrib/python-portend/template.py:11]: turns out it's tempora runtime dep

**Pull request closed without merging**